src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -265,155 +313,356 @@ export type RemoteConfig = {
+		end
+
 		self:LogPlayerEvent({
 			userId = player.UserId,
 			event = data.event,
-			value = data.value
+			value = data.value,
 		})
 	end)
-	
+
 	-- Logs an event to be sent once the player is leaving the game
 	self.Client.AddDelayedEvent:Connect(function(player: Player, data: DelayedEvent)
+		if data == nil or type(data) ~= "table" then
+			return
+		end
+
+		if data.event == nil then
+			return
+		end
+
+		if type(data.event) ~= "string" then
+			return
+		end
+
+		if data.userId ~= nil and type(data.userId) ~= "number" then
+			return
+		end
+
+		if data.value ~= nil and type(data.value) ~= "number" then
+			return
+		end
+
 		self:AddDelayedEvent({
 			userId = player.UserId,
 			event = data.event,
-			value = data.value
+			value = data.value,
 		})
 	end)
-	
+
 	-- Adds a value to a tracked event, it will be sent once the player is leaving the game
 	self.Client.AddTrackedValue:Connect(function(player: Player, data: TrackedValueEvent)
+		if data == nil or type(data) ~= "table" then
+			return
+		end
+
+		if data.event == nil then
+			return
+		end
+
+		if type(data.event) ~= "string" then
+			return
+		end
+
+		if data.userId ~= nil and type(data.userId) ~= "number" then
+			return
+		end
+
+		if data.value ~= nil and type(data.value) ~= "number" then
+			return
+		end
+
 		self:AddTrackedValue({
 			userId = player.UserId,
 			event = data.event,
-			value = data.value
+			value = data.value,
+		})
+	end)
+
+	self.Client.SetCustomDimension:Connect(function(player: Player, data: DimensionEvent)
+		if data == nil or type(data) ~= "table" then
+			return
+		end
+
+		if data.value == nil or data.dimension == nil then
+			return
+		end
+
+		if type(data.value) ~= "string" then
+			return
+		end
+
+		if type(data.dimension) ~= "string" then
+			return
+		end
+
+		self:SetCustomDimension({
+			userId = player.UserId,
+			dimension = data.dimension,
+			value = data.value,
 		})
 	end)
-	
+
 	MarketplaceService.PromptBundlePurchaseFinished:Connect(
 		function(player: Player, bundleId: number, purchased: boolean)
 			if not purchased then
 				return
 			end
-			
+
+			local productInfo = self:_getProductInfo(bundleId, Enum.InfoType.Bundle)
+
 			self:LogMarketplacePurchase({
 				userId = player.UserId,
 				itemType = "Bundle",
-				id = bundleId,
+				id = productInfo and productInfo.Name or bundleId,
 				cartType = "PromptPurchase",
+				robuxPrice = productInfo and productInfo.PriceInRobux,
 			})
 		end
 	)
-	
+
 	MarketplaceService.PromptGamePassPurchaseFinished:Connect(
 		function(player: Player, gamePassId: number, purchased: boolean)
 			if not purchased then
 				return
 			end
-			
+
+			local productInfo = self:_getProductInfo(gamePassId, Enum.InfoType.GamePass)
+
 			self:LogMarketplacePurchase({
 				userId = player.UserId,
 				itemType = "GamePass",
-				id = gamePassId,
+				id = productInfo and productInfo.Name or gamePassId,
 				cartType = "PromptPurchase",
+				robuxPrice = productInfo and productInfo.PriceInRobux,
```

Comment:
Is LogMarketplacePurchase designed to accept robuxPrice as nil? If productInfo is nil for some reason, robuxPrice will be nil instead of 0.